lib/atomic_bool.go

```diff
@@ -0,0 +1,25 @@
+package lib
+
+import (
+	"sync/atomic"
+)
+
+type AtomicBool int32
+
+func NewAtomicBool(value bool) *AtomicBool {
```

Comment:
Is this abstraction worthwhile? `foo.Flag.Set()` and `foo.Flag.IsSet` don't seem like an obviously huge improvement over `atomic.StoreInt32(&foo.Flag, 1)` etc. I buy that they state an intention more clearly but I'm just not convinced there is enough value to abstract this. I can be convinced though 😄 .

What's here is fine and simple but there are no docs which for a "library" method isn't ideal and it doesn't quite cover the whole interface we need in general. In Connect code in a couple of places we use atomic bools for stop flags and it's important in that pattern that you can not only flip the flag to set it but also know if was set before atomically to do one-time clean up tasks like closing chans. I use `atomic.SwapInt32` or similar there so would be nice to have that here too if we want to re-use this generally?

Could even just make `Set()` into `Set() bool` and have it return whether or not it was set _before_ the operation?

Reply:
I thought it was better when I was originally writing it. Using int32s and having `atomic.StoreInt32` + imports of "sync/atomic" didn't seem quite as clean as just having the type being stored know how to set and check itself.

In the end I isolated usage of the variables that I was creating as the AtomicBool so it probably isn't worth keeping.